src/assets.zig

```diff
@@ -42,76 +89,117 @@ fn readDefaultFile(allocator: NeverFailingAllocator, dir: std.fs.Dir) !ZonElemen
 	return .null;
 }
 
-/// Reads .zig.zon files recursively from all subfolders.
-pub fn readAllZonFilesInAddons(externalAllocator: NeverFailingAllocator, addons: main.List(std.fs.Dir), addonNames: main.List([]const u8), subPath: []const u8, defaults: bool, output: *std.StringHashMap(ZonElement)) void {
+/// Reads all asset `.zig.zon` files recursively from all sub folders.
+///
+/// Files red are stored in output hashmap with asset ID as key.
+/// Asset ID are constructed as `{addonName}:{relativePathNoSuffix}`.
+/// relativePathNoSuffix is always unix style path with all extensions removed.
+/// `output` and `migrations` must share same `externalAllocator`.
```

Comment:
Please use asserts for cases like this.
A doc comment cannot be automatically checked, and doc comments tend to decay over time as changes are made to the code.